app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -158,7 +158,7 @@ class SubmissionFileManager extends React.Component {
           />
         </div>
       );
-    } else {
+    } else if (!this.props.is_solution_file){
```

Comment:
If you're trying to disable the file viewer, don't call the renderFileViewer function at all. Otherwise there is going to be the fieldset/legend/span etc. rendered without any content.